skate-plugin/src/main/kotlin/com/slack/sgp/intellij/featureflags/PsiElementExtensions.kt

```diff
@@ -16,34 +16,31 @@
 package com.slack.sgp.intellij.featureflags
 
 import com.intellij.psi.PsiElement
+import com.intellij.psi.impl.source.tree.LeafPsiElement
 import com.intellij.psi.util.childrenOfType
-
-fun PsiElement.getFirstChild(elementType: String): PsiElement? {
-  return this.childrenOfType<PsiElement>().firstOrNull {
-    it.node.elementType.toString() == elementType
-  }
-}
-
-fun PsiElement.getChildren(elementType: String): List<PsiElement> {
-  return this.childrenOfType<PsiElement>().filter { it.node.elementType.toString() == elementType }
-}
-
-fun PsiElement.getAnnotationEntries(): List<PsiElement> {
-  return this.getFirstChild("MODIFIER_LIST")?.getChildren("ANNOTATION_ENTRY") ?: listOf()
-}
-
-fun PsiElement.getAnnotationName(): String? {
-  return this.getFirstChild("CONSTRUCTOR_CALLEE")?.text
+import org.jetbrains.kotlin.lexer.KtTokens
+import org.jetbrains.kotlin.psi.KtAnnotationEntry
+import org.jetbrains.kotlin.psi.KtEnumEntry
+import org.jetbrains.uast.UExpression
+import org.jetbrains.uast.toUElement
+
+fun KtEnumEntry.hasAnnotation(name: String): Boolean {
+  return this.getAnnotationEntries().any { it.shortName?.asString() == name }
 }
 
-fun PsiElement.isEnum(): Boolean {
-  return this.node?.elementType?.toString() == "ENUM_ENTRY"
+fun KtEnumEntry.getAnnotation(name: String): KtAnnotationEntry? {
+  return this.getAnnotationEntries().firstOrNull { it.shortName?.asString() == name }
 }
 
-fun PsiElement.hasAnnotation(name: String): Boolean {
-  return this.getAnnotationEntries().any { it.getAnnotationName() == name }
+fun PsiElement.getEnumIdentifier(): LeafPsiElement? {
+  return this.childrenOfType<LeafPsiElement>().firstOrNull { it.elementType == KtTokens.IDENTIFIER }
 }
 
-fun PsiElement.getEnumIdentifier(): PsiElement? {
-  return this.getFirstChild("IDENTIFIER")
+fun KtAnnotationEntry.getKeyArgumentValue(key: String): String? {
+  val argumentExpression =
+    this.valueArguments
+      .firstOrNull { it.getArgumentName()?.asName?.asString() == key }
+      ?.getArgumentExpression()
+  return (argumentExpression as? PsiElement)?.toUElement(UExpression::class.java)?.evaluate()
```

Comment:
hmmm you shouldn't need to do this first `as? PsiElement`, can you find another way?

Reply:
I did `as? PsiElement` to make compiler happy as without it I get following error. So far the only way I found to evaluate() expression was to convert the element to UExpression.

```
e: file:///Users/arpita.patel/slack-gradle-plugin/skate-plugin/src/main/kotlin/com/slack/sgp/intellij/featureflags/PsiElementExtensions.kt:40:30 Unresolved reference. None of the following candidates is applicable because of receiver type mismatch: 
public fun PsiElement?.toUElement(): UElement? defined in org.jetbrains.uast
```

Reply:
[ec81f44](https://github.com/slackhq/slack-gradle-plugin/pull/576/commits/ec81f4417cd0409bed9ab368d3bdb606b12ea717)
Thanks for suggestions and hint that I should just covert the PsiFile directly to UFile early and start using UAST APIs directly from the start instead of `as? PsiElement` and other lower level API calls first